SystemInformer/mainwnd.c

```diff
@@ -100,17 +100,79 @@ BOOLEAN PhMainWndInitialization(
         PhInitializeStringBuilder(&stringBuilder, 100);
         PhAppendStringBuilder2(&stringBuilder, PhApplicationName);
 
-        if (currentUserName = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+        if (!PhGetIntegerSetting(L"EnableVerboseWindowTitle"))
         {
-            PhAppendStringBuilder2(&stringBuilder, L" [");
-            PhAppendStringBuilder(&stringBuilder, &currentUserName->sr);
-            PhAppendCharStringBuilder(&stringBuilder, L']');
-            if (KphIsConnected()) PhAppendCharStringBuilder(&stringBuilder, L'+');
-            PhDereferenceObject(currentUserName);
+            if (currentUserName = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+            {
+                PhAppendStringBuilder2(&stringBuilder, L" [");
+                PhAppendStringBuilder(&stringBuilder, &currentUserName->sr);
+                PhAppendCharStringBuilder(&stringBuilder, L']');
+                if (KphIsConnected()) PhAppendCharStringBuilder(&stringBuilder, L'+');
+                PhDereferenceObject(currentUserName);
+            }
+
+            if (PhGetOwnTokenAttributes().ElevationType == TokenElevationTypeFull)
+                PhAppendStringBuilder2(&stringBuilder, L" (Administrator)");
         }
+        else
+        {
+            PhAppendFormatStringBuilder(&stringBuilder, L" %lu", HandleToUlong(NtCurrentProcessId()));
+
+            BOOL isRunningAsSystem = FALSE;
+
+            if (currentUserName = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+            {
+                PPH_STRING strSystemUsername = PhCreateString(L"NT Authority\\System");
+                isRunningAsSystem = (PhCompareString(currentUserName, strSystemUsername, TRUE) == 0);
```

Comment:
You're performing a LSA lookup for a SID to just get a string for comparison? `RtlEqualSid` is the way to go.